models/repo.go

```diff
@@ -1275,6 +1275,20 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// Create/Remove git-daemon-export-ok for git-daemon...
+		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
+		if repo.IsPrivate {
+			if err = os.Remove(daemonExportFile); err != nil {
```

Comment:
that'd work, though checking if a file exists before removing it is bad practice... since it can be added/removed in between the check and the removal...

also this would break
```
if repo.IsPrivate && com.IsExist {
 [...]
} else { // if file doesn't exist, this block is ran even if it's private...
  os.Create(...)
}
```

So in that case I'd have to do:
```
if repo.IsPrivate {
  if com.IsExist {
    if os.Remove(...)
[...]
```
Which doesn't look very nice :sweat: